lib/constructs/vpc-default-sg-rule-remove.ts

```diff
@@ -0,0 +1,57 @@
+import * as cdk from "aws-cdk-lib";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import * as iam from "aws-cdk-lib/aws-iam";
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import * as nodejs from "aws-cdk-lib/aws-lambda-nodejs";
+import * as cr from "aws-cdk-lib/custom-resources";
+import { Construct } from "constructs";
+
+export interface VpcDefaultSecurityGroupRuleRemoverProps {
+  // This has to accept the concrete class because we need the VPC and
+  // attributes only available on the concrete class.
+  vpc: ec2.Vpc;
+}
+export class VpcDefaultSecurityGroupRuleRemover extends Construct {
+  constructor(scope: Construct, id: string, props: VpcDefaultSecurityGroupRuleRemoverProps) {
+    super(scope, id);
+
+    const handler = new nodejs.NodejsFunction(this, "VpcSgRuleRemovalHandler", {
+      runtime: lambda.Runtime.NODEJS_16_X,
+      entry: "lib/custom-resources/delete-default-sg-rules.ts",
+      handler: "onEvent",
+      vpc: props.vpc,
+      initialPolicy: [
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: ["ec2:DescribeSecurityGroupRules"],
+          // Describe has to be allowed on all VPCs in order to perform the filter
+          resources: ["*"],
+        }),
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: ["ec2:RevokeSecurityGroupIngress", "ec2:RevokeSecurityGroupEgress"],
+          // Only allow changing rules on the default security group
+          resources: [
+            cdk.Stack.of(this).formatArn({
+              service: "ec2",
+              resource: "security-group",
+              resourceName: props.vpc.vpcDefaultSecurityGroup,
```

Comment:
Just making sure since you mentioned the criticality of passing in the right SG - are you saying that we need to make sure that this line never changes to some other SG? Also, I don't see that property in  https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.VpcProps.html but maybe that's not the right place to be looking?

Reply:
Yeah so `Prop` objects are usually inputs. The type is defined on https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html#vpcdefaultsecuritygroup but notably not on `ec2.IVpc` (which is why the `Props` for this construct takes `ec2.Vpc` instead of `ec2.IVpc`).

Reply:
OK got it, thanks. Not that they have any actual description in the docs but at least it confirms the field name.

Reply:
Yeah... https://github.com/aws/aws-cdk/blob/1485f48c515f81fa585a325e02f1602f56443605/packages/%40aws-cdk/aws-ec2/lib/vpc.ts#L1390 is where the implementation itself is, matching https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpc.html#DefaultSecurityGroup-fn::getatt